pkg/util/limiter/query_limiter.go

```diff
@@ -93,7 +98,11 @@ func (ql *QueryLimiter) AddSeries(seriesLabels []mimirpb.LabelAdapter) error {
 
 	ql.uniqueSeries[fingerprint] = struct{}{}
 	if len(ql.uniqueSeries) > ql.maxSeriesPerQuery {
-		// Format error with max limit
+		if len(ql.uniqueSeries) == ql.maxSeriesPerQuery+1 {
+			// If we've just exceeded the limit for the first time for this query, increment the failed query metric.
+			ql.queryMetrics.QueriesExceededLimits.WithLabelValues("max-fetched-series-per-query").Inc()
```

Comment:
[nit] I'm wondering if really need the `-per-query` suffix here and in other limits too. Given this metric tracks the limits per query "by definition", we could probably omit the suffix.

Reply:
I wen't with the `-per-query` suffix to keep the name the same as the CLI option (and therefore make it easy and unambiguous to find in the docs). I'm happy either way though.